File: Data Structures & Algorithms/maximum-subarray/submission-2.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = nums[0];
        int cur = 0;

        for ( int right = 0; right < nums.size(); right++ ) {
            cur += nums[right];
            sum = max(sum, cur);

            if ( cur < 0 ) {
                cur = 0;
            }
        }
        return sum;
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        height = 0

        if root == None:
            return 0
        
        else:
            height = 1 + max(self.maxDepth(root.right), self.maxDepth(root.left))

        return height